### PR TITLE
Extend multi-cluster with VM testing

### PR DESCRIPTION
**Pull request by ZhengzheYang**
This PR aims to add VM to the multicluster testing. 
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/multicluster/helper.go [hunk 1]
```diff
@@ -47,6 +47,10 @@ func Setup(controlPlaneValues *string, clusterLocalNS, mcReachabilityNS *namespa
 
 		// Set the cluster-local namespaces in the mesh config.
 		*controlPlaneValues = fmt.Sprintf(`
+values:
+  global:
+    meshExpansion:
+      enabled: true
```
Comment (stevenctl) [inline tests/integration/multicluster/helper.go:53]: I'm not sure if this is necessary, it should be enabled as part of the multicluster install https://github.com/istio/istio/blob/bb809f91096d9abf33a69b532d1c2cf4338099f6/pkg/test/framework/components/istio/operator.go#L436

**Comment by stevenctl** [comment]
As part of #24549, the [meshnetworks](https://github.com/istio/istio/pull/24549/files#diff-a6eba6ceb510a5fd82c18dacb28ead2e) test may already give us coverage here. In general I think it's best to avoid adding tests here that don't check scenarios that aren't exclusive to multicluster setups.

File: tests/integration/multicluster/helper.go (hunk 1 above)
Comment (ZhengzheYang) [reply tests/integration/multicluster/helper.go:53]: Yes, I am just experimenting around. This is far from ready. Thanks for pointing it out, though!

**Comment by ZhengzheYang** [comment]
> As part of #24549, the [meshnetworks](https://github.com/istio/istio/pull/24549/files#diff-a6eba6ceb510a5fd82c18dacb28ead2e) test may already give us coverage here. In general I think it's best to avoid adding tests here that don't check scenarios that aren't exclusive to multicluster setups.

I totally agree with you. This is somewhat of an experiment PR to learn more about how VM should work with multicluster. I agree that the tests you are covering in your PR is comprehensive enough. If I were to add any new tests, I will request a review from you. Thanks!

**Comment by stevenctl** [comment]
> As part of #24549, the meshnetworks test may already give us coverage here. 

Well.. turns out we're just checking discovery and not true reachability there. That said, if we have reachability tests for VMs for single-cluster, it will be better to convert the existing tests to run in multicluster as well. 

I'm actively working on scoping down that PR to a "base" to allow tests to be incrementally converted. When that's done we can start adding coverage for single & multicluster in one go. 

There is a hidden prow job you can trigger with `/test integ-pilot-multicluster-tests_istio` to run the pilot tests but with the same cluster setup as the multicluster job.

**Comment by ZhengzheYang** [comment]
Understood! Thanks for the info.

**Pull request closed without merging**